python/lsst/ip/diffim/imageDecorrelation.py

```diff
@@ -0,0 +1,191 @@
+from __future__ import absolute_import, division, print_function
+#
+# LSST Data Management System
+# Copyright 2008-2016 AURA/LSST.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <https://www.lsstcorp.org/LegalNotices/>.
+#
+
+import numpy as np
+from scipy.stats import sigmaclip
+from scipy.fftpack import fft2, ifft2, ifftshift
+from scipy.ndimage.filters import convolve
+
+import lsst.afw.image as afwImage
+import lsst.meas.algorithms as measAlg
+import lsst.afw.math as afwMath
+
+__all__ = ("performExposureDecorrelation")
+
+
+def computeClippedImageStats(im, low=3, high=3):
+    """! Utility function for sigma-clipped array statistics.
+    @param im   A numpy array. Should work on any dimensions/shape.
+    @param low  Lower bound factor of sigma clipping. Default is 3.
+    @param high  Upper bound factor of sigma clipping. Default is 3.
+    @return sigma-clipped mean and std of input array
+    """
+    _, low, upp = sigmaclip(im, low=low, high=high)
+    tmp = im[(im > low) & (im < upp)]
+    mean1 = np.nanmean(tmp)
+    sig1 = np.nanstd(tmp)
+    return mean1, sig1
+
+
+def getImageGrid(im):
+    """! Convenience function for creating a grid of pixel coordinates covering a 2-d array,
+    using `numpy.meshgrid()`
+    @param im input 2-d numpy array
+    @return two numpy arraysm each with same shape as `im`, containing x- and y- coordinate
+    grid.
+    """
+    xim = np.arange(np.int(-np.floor(im.shape[0]/2.)), np.int(np.floor(im.shape[0]/2)))
+    yim = np.arange(np.int(-np.floor(im.shape[1]/2.)), np.int(np.floor(im.shape[1]/2)))
+    x0im, y0im = np.meshgrid(xim, yim)
+    return x0im, y0im
+
+
+def computeDecorrelationCorrectionKernel(kappa, sig1=0.2, sig2=0.2):
+    """! Compute the Lupton/ZOGY post-conv. kernel for decorrelating an
+    image difference, based on the PSF-matching kernel.
+    @param kappa  A matching kernel array derived from Alard & Lupton PSF matching
+    @param sig1   Average sqrt(variance) of template image used for PSF matching
+    @param sig2   Average sqrt(variance) of science image used for PSF matching
+    @return a 2-d numpy.array containing the correction kernel
+
+    @note As currently implemented, kappa is a static (single) kernel.
+    """
+    def post_conv_kernel_ft2(kernel, sig1=1., sig2=1.):
```

Comment:
Also, I am puzzled why you have these two functions instead of just the one below. This one seems too small to stand on its own and I don't think any other code is calling it.